Document custom session parameters via env variables

lucas-koehler opened the pull request.

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+## Adding environment variables via REST API
+
+The simplest way to start a Session with custom environment variables is using Theia Cloud's REST API.
+Theia Cloud offers npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
```
```suggestion
Theia Cloud offers an npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
```

jfaltermeier replied:
Sorry, I read the markdown formatting as actual brackets :smile:

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+Theia Cloud offers npm package [@eclipse-theiacloud/common](https://www.npmjs.com/package/@eclipse-theiacloud/common) to access the REST API conveniently.
+It offers to provide environment variables directly or as lists of [config maps](https://kubernetes.io/docs/concepts/configuration/configmap/) or [secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to read in.
+
+The following code snippet show how to add environment variables while starting a session.
```
```suggestion
The following code snippet shows how to add environment variables while starting a session.
```

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+import { getTheiaCloudConfig, SessionStartRequest, TheiaCloud } from '@eclipse-theiacloud/common';
+
+// Get common parameters from Theia Cloud configuration
+const { appDefinition, appId, serviceUrl} = getTheiaCloudConfig();
```
```suggestion
const { appDefinition, appId, serviceUrl } = getTheiaCloudConfig();
```

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+
+The environment variables can also be configured directly through the Session custom resource by adding one or multiple of the following properties:
+
+- [envVars](https://github.com/eclipsesource/theia-cloud-helm/blob/4cd9d98d30cebe8d31e7084369878c1c2d28776c/charts/theia-cloud-crds/templates/session-spec-resource.yaml#L46-L49): This property is a map that allows you to define environment variables directly. Each entry in the map consists of the environment variable name and its value. This method is suitable for standard configuration needs and ...
```
I don't like the links to specific commits, however I don't see a better option. 
Maybe we should keep track where such links are used, so we may check them regularly?

lucas-koehler replied:
Mhm, yes I agree that it is not optimal to use it like that. If we point to main, we would also need to keep track and check links regularly. Especially with references to CRDs where a new version will throw off the referenced lines completely.

Maybe we could keep a file like `repository-links` in the root of the repository that either keeps entries for every link and its file or just files that contain repository links. What do you think?

jfaltermeier replied:
Sounds good, I think keeping the files should be enough. 
I'll add it to the release template to check for outdated links after a release

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+    fromMap: {
+      mykey: "myvalue"
+    },
+    fromConfigMaps: [ "session-config-map-1", "config-map-2"],
```
```suggestion
    fromConfigMaps: [ "session-config-map-1", "config-map-2" ],
```

jfaltermeier commented on content/documentation/sessionParameters/content.md:
```diff
@@ -0,0 +1,84 @@
+      mykey: "myvalue"
+    },
+    fromConfigMaps: [ "session-config-map-1", "config-map-2"],
+    fromSecrets: [ "session-secret-1", "session-secret-2"]
```
```suggestion
    fromSecrets: [ "session-secret-1", "session-secret-2" ]
```

jfaltermeier reviewed (commented):
Already looks very good to me, I have a few smaller suggestions though.

lucas-koehler commented:
@jfaltermeier I added the `repository-links` file and added the content.md for the session parameters. For now, I left in the link to a specific commit because it makes it easier to see what should be referenced even if the file changes on main.
